ROADMAP.md

```diff
@@ -0,0 +1,123 @@
+
+  *How will we achieve this?*<br>
+  By gradually adding more unit and functional tests and fixing issues when the opportunity arises. We won't have long stints where we focus exclusively on this goal, but it's something we'll dedicate as much time as possible to.
+
+  *Definition of Done*<br>
+  The goal is not to reach 100% code coverage, but realistically speaking 80 or 90% should be achievable. Our CI test runs should be stable and flake-free and we should have no linter issues.
+
+
+Long-term goals
+---------------
+
+These are goals achievable after a year, tentatively done in 2024 and beyond.
+
+- **Add support for Firefox and other WebKit-based browsers.**<br>
+  Currently (April 2022) our main focus is supporting Chromium based browsers. This should be expanded to include other browsers as well. The main challenges here will be around CDP and the differences in behavior between browsers.
+
+  *How will we achieve this?*<br>
+  By testing other browsers and fixing issues as they arise.
+
+  *Definition of Done*<br>
+  When the latest Firefox and certain WebKit-based browsers are as well supported as Chromium.
+
+
+- **Feature parity with Playwright and Puppeteer.**<br>
+  Currently our functionality is limited compared to more mature projects like Playwright and Puppeteer. We plan to expand this gradually and reach or exceed the features offered by other browser automation tools.
+
+  *How will we achieve this?*<br>
+  By prioritizing new features to add based on API importance and user feedback. After the short-term stability improvements are done, our main focus will be to add more missing features and close the feature gap we currently have.
+
+  *Definition of Done*<br>
+  When all of the functionality found in other tools is implemented in xk6-browser.
+
+
+- **Merge extension into k6 core.**<br>
+  k6 and Grafana consider browser testing to be another strategy under the testing umbrella, along with performance, contract and other types of testing modern web applications benefit from. As such, the scope of the k6 tool will expand to include functional/E2E testing using browser automation, with the ultimate goal of merging the xk6-browser extension into the main k6 repository as a core module.
+
+  *How will we achieve this?*<br>
+  Once the extension is stable, feature complete, and widely used in k6 Cloud, we will create a PR to the main k6 repository with the extension code. The standalone xk6-browser repository will be deprecated, and users will be pointed to use the main k6 binary instead.
+
+  This will be minor a breaking change, as scripts will have to import `k6/browser` instead of `k6/x/browser` and CI jobs will need to change, but the functionality should remain the same.
+
+  *Definition of Done*<br>
+  Once the extension has been merged into k6 core.
```

Comment:
```suggestion
- **Merge the extension into the k6 tool.**<br>
  k6 and Grafana consider browser testing another strategy under the testing umbrella, along with performance, contract, and other types of testing modern web applications benefit from. As such, the scope of the k6 tool will expand to include functional/E2E testing using browser automation, with the ultimate goal of merging the xk6-browser extension into the k6 tool.

  *How will we achieve this?*<br>
  Once the extension is stable, feature-complete, and widely used in k6 Cloud, we will create a PR to the main k6 repository with the extension code. Then we will deprecate the standalone xk6-browser repository and point users to use the main k6 binary instead.

  This will be a minor breaking change, as scripts will have to import `k6/browser` instead of `k6/x/browser`, and CI jobs will need to change, but the functionality should remain the same.

  *Definition of Done*<br>
  Once the extension has been merged into the k6 tool.
```

Please correct me if I'm wrong, but k6 core is about running and managing VUs, dealing with Goja, async stuff, etc. However, even if they are part of the k6 tool, the modules are, well, modules.

Reply:
In the context of extensions, we usually refer to "k6 core" as the "main k6 repository", and don't make a distinction between "core k6 functionality" and "k6 JS modules".

I changed it slightly in 89dd838. Let me know if it's clearer now.

Reply:
It's crystal clear.